sys/bus.c

```diff
@@ -98,3 +98,55 @@ int bus_generic_probe(device_t *bus) {
   }
   return 0;
 }
+
+// identify hinted devices and resources.
+// add devices to parentbus
+// add hinted devices to devices ivars
+/*
+int bus_enumerate_hinted_children(device_t *bus){
```

Comment:
I would prefer to make a change with device hints in a separate PR - otherwise the PR contents 1) might be confusing - ideally one PR should introduce one concept, devclasses and device hints are separate ideas that does not interfere with each other *that much* 2) the less number of files PR touches the better, by joining these two ideas we might do a bunch of unrelated stuff - python scripts, FDT files along with devclasses structure

What do you think?

Reply:
Let's discuss that face-to-face. It's complicated.